src/dagmc/dagmcmetadata.cpp

```diff
@@ -0,0 +1,591 @@
+#include "dagmcmetadata.hpp"
+#include <iostream>
+#include <set>
+#include <algorithm>
+
+// constructor for metadata class
+dagmcMetaData::dagmcMetaData(moab::DagMC* dag_ptr, bool verbosity) {
+  DAG = dag_ptr; // dagmc pointer
+  verbose = verbosity;
+  // these are the keywords that dagmc will understand
+  // from groups if you need to process more
+  // they should be added here
+  metadata_keywords.push_back("mat");
+  metadata_keywords.push_back("rho");
+  metadata_keywords.push_back("boundary");
+  metadata_keywords.push_back("tally");
+  metadata_keywords.push_back("importance");
+
+  // allow some synonyms
+  keyword_synonyms[ "rho" ] = "density";
+  keyword_synonyms[ "mat" ] = "material";
+}
+
+// destructor doesnt need to do anything
+dagmcMetaData::~dagmcMetaData() {
+}
+
+// load the property data from the dagmc instance
+void dagmcMetaData::load_property_data() {
+  parse_material_data();
+  parse_importance_data();
+  parse_boundary_data();
+  parse_tally_volume_data();
+  parse_tally_surface_data();
+  // finalise_counters();
+}
+
+// get the a given volume property on a given entity handle
+std::string dagmcMetaData::get_volume_property(std::string property, moab::EntityHandle eh) {
+  std::string value = "";
+  if (property == "material_density") {
+    value = volume_material_property_data_eh[eh];
+  } else if (property == "material") {
+    value = volume_material_data_eh[eh];
+  } else if (property == "density") {
+    value = volume_density_data_eh[eh];
+  } else if (property == "importance") {
+    value = volume_importance_data_eh[eh];
+  } else if (property == "tally") {
+    value = tally_data_eh[eh];
+  } else {
+    std::cout << "Not a valid property for volumes" << std::endl;
+  }
+  return value;
+}
+
+// overloaded get_volume_property for indices and id's'
+std::string dagmcMetaData::get_volume_property(std::string property, int vol, bool idx) {
+  // if this is an index query
+  moab::EntityHandle eh;
+  if (idx == true) {
+    eh = DAG->entity_by_index(3, vol);
+  } else {
+    eh = DAG->entity_by_id(3, vol);
+  }
+  return get_volume_property(property, eh);
+}
+
+// Get a given property on a surface
+std::string dagmcMetaData::get_surface_property(std::string property, moab::EntityHandle eh) {
+  std::string value = "";
+  if (property == "boundary") {
+    value = surface_boundary_data_eh[eh];
+  } else if (property == "tally") {
+    value = tally_data_eh[eh];
+  } else {
+    std::cout << "Not a valid property for surfaces" << std::endl;
```

Comment:
Best Practices: I would recommend sending this to `std::cerr` rather than `std::cout`.  Additionally, I would recommend printing the invalid property in the error, to help the user with the invalid input.